### PR TITLE
Yieldlab Bid Adapter: Add Digital Services Act (DSA) handling

### PR DESCRIPTION
**Pull request by nkloeber**
## Type of change

- [x] Feature

## Description of change

Updated the Yieldlab Bid Adapter to handle Digital Services Act (DSA) parameters in both bid requests and responses. 
## Other information

A review and approval from a member of the Yieldlab team are required. (@brushmate, @rey1128, @Mufas61 or @kippsterr)

### CODE REVIEW

**Review by dgirardi — COMMENTED** [review]
My understanding on how DSA will work with Prebid:

 - the pub sets `ortb2.regs.ext.dsa`, not bid params. your adapter can find it in `bidderRequest.ortb2`
 - DSA info in bid responses should go under `meta.dsa` (not `ext.dsa`)

CC @patmmccann

File: modules/yieldlabBidAdapter.js [hunk 1]
```diff
@@ -83,6 +80,29 @@ export const spec = {
       if (floor) {
         adslotFloors.push(bid.params.adslotId + ':' + floor);
       }
+      const dsaRequired = bid.params.regs?.ext?.dsa?.required;
```
Comment (patmmccann) [inline modules/yieldlabBidAdapter.js:83]: you need to follow the spec; please look at the appnexus and magnite pr's for the location of dsa information in the bid request object.

**Review by patmmccann — CHANGES_REQUESTED** [review]

File: modules/yieldlabBidAdapter.js (hunk 1 above)
Comment (nkloeber) [reply modules/yieldlabBidAdapter.js]: Thank you, I updated the pr.

**Comment by patmmccann** [comment]
please see https://github.com/InteractiveAdvertisingBureau/openrtb/pull/158/files

**Comment by rey1128** [comment]
LGTM, we now also follow the newly updated spec for `dsarequired` and `dsaparams`

**Review by patmmccann — APPROVED** [review]

**Pull request merged**